docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,164 @@
+
+## 3. Keep the data alive
+The evolution of RAG toward a communication system means that the data is no longer just a passive reference stock. It has to be kept alive in two ways: continuously transformed and reshaped to better fit the retrieval objective, constantly circulating across different flows.
+
+### 3.1 You need bad data
+The reality of production RAG is that once we get slightly beyond the proof-of-concept phase, the system has to work with whatever data is available. To get to the point, you have to become an expert on bad data.
+
+Scaling is not only a bitter lesson for LLMs: I worked for years in cultural heritage and a library with millions of bad OCR documents is always considerably more useful than a well curated and edited selection of a thousand canon texts. Large organizations are no different. RAG has to work with legacy texts, which means PDFs. And, to stress it again, LLMs are hardly trained on PDFs: common crawl is web/html texts, without artifact and advanced document structures. Bad data is also to be expected on the input side. Users want the best answers with the minimum efforts, and the customer is always right. 
+
+In both cases, LLM trainers do not necessarily have access to the actual bad data processed in production. It can be restricted due to data protection regulations or other organizational concerns, or simply hard to access. The main alternative is looking for placeholders or… generating it. As it turns out Google and probably other major AI providers have been seriously investing in data degradation techniques. Simple data processing heuristics can be already effective (like dropping cases, introducing noise). The most powerful technique I’ve yet found is to fine-tune on existing bad data.
+
+### 3.2 Managing data flows with classifiers
+Fortunately, we have the best tools to date for bad (and good) data. Despite their limitations for search, language models are amazing at data processing. I mean here language models in general, not just the popular decoders: the previous generation of encoders and encoder-decoders currently enjoy a much needed revival; they provide a much better parameters / performance / robustness ratio than GPT-4, Llama or Mistral for many data tasks.
+
+Classifiers and reformulators (for want of a better name) are increasingly critical components of my RAG systems. This include:
```

Comment:
"my" = "your", correct?

Reply:
No really rather mine, but better phrase. "I increasingly rely on classifiers or reformulators (for want of better name) to build RAG systems. This includes:"